tests/unit/test_project.py

```diff
@@ -197,6 +197,26 @@ def test_project_unmarshal_with_unsupported_fields(unsupported_field, yaml_loade
     )
 
 
+@pytest.mark.parametrize(
+    "variable,is_forbidden",
+    [("$BAR", True), ("BAR_$BAZ", True), ("BAR$", False)],
```

Comment:
How ``BAR$`` is not forbidden ?

Reply:
why should it be? if it is the last char, then it is just a regular char.

Reply:
Yeah true,  sorry

Reply:
;)